src/main/scala/org/constellation/API.scala

```diff
@@ -229,14 +236,33 @@ class API(udpAddress: InetSocketAddress)(implicit system: ActorSystem,
     post {
       pathPrefix("channel") {
         path("open") {
-          entity(as[ChannelOpenRequest]) { request =>
-            ChannelMessage.createGenesis(request)
-            complete(StatusCodes.Created)
+          entity(as[ChannelOpen]) { request =>
+            onComplete(
+              ChannelMessage.createGenesis(request)
+            ) { res =>
+              complete(res.getOrElse(ChannelOpenResponse("API Failure")).json)
+            }
+
           }
         } ~
           path("send") {
             entity(as[ChannelSendRequest]) { send =>
-              complete(ChannelMessage.createMessages(send))
+              onComplete(ChannelMessage.createMessages(send)) { res =>
+                complete(res.getOrElse(ChannelOpenResponse("API Failure")).json)
+              }
+            }
+          } ~
+          path("send" / "json") {
+            entity(as[ChannelSendRequestRawJson]) { send: ChannelSendRequestRawJson =>
+              import constellation._
+              val amended = ChannelSendRequest(
+                send.channelId,
+                send.messages.x[Seq[JValue]].map { _.json }
```

Comment:
what is going on here? if the messages are already json strings... why extract them and convert them back? is this to ensure they are valid json?

Reply:
Has to do with how they're sent from the UI. They're grabbed from a textarea as a single string of json, but they need to be changed to Seq[String] for the create function.